File: OP_Lab5/py/tvector2d.py
from tvector import TVector
from math import sqrt


class TVector2D(TVector):
    def __init__(self, x, y):
        self.__x = x
        self.__y = y

    def getScalar(self, other):
        return self.__x * other.getCoordinate("x") + self.__y * other.getCoordinate("y")

    def getLength(self):
        return sqrt(self.__x * self.__x + self.__y * self.__y)

    def getCoordinate(self, c):
        if c == "x":
            return self.__x
        elif c == "y":
            return self.__y
        else:
            return 0

    def print(self, header):
        print(f"{header} TVector2D: x = {self.__x:5} y = {self.__y:5}")

File: OP_Lab5/cpp/tvector2d.cpp
#include "tvector2d.h"

double TVector2D::getCoordinate(char c)
{
	switch (c)
	{
	case 'x':
		return x;
		break;
	case 'y':
		return y;
		break;
	}
}

double TVector2D::getScalar(TVector& vector) {
	return x * vector.getCoordinate('x') + y * vector.getCoordinate('y');
}

double TVector2D::getLength() {
	return sqrt(x * x + y * y);
}

void TVector2D::print(string s) {
	cout << s << "TVector2D: " << " x = " << setw(4) << x << " y = " << setw(4) << y << endl;
}

